src/modal/modal-backdrop.ts

```diff
@@ -1,5 +1,29 @@
-import {Component} from '@angular/core';
+import {ChangeDetectorRef, Component} from '@angular/core';
+import {trigger, transition, AnimationEvent, useAnimation} from '@angular/animations';
+import {Subject} from 'rxjs/Subject';
+import {fadeIn, fadeOut} from '../animations/fade';
 
-@Component({selector: 'ngb-modal-backdrop', template: '', host: {'class': 'modal-backdrop fade show'}})
+@Component({
+  selector: 'ngb-modal-backdrop',
+  template: '',
+  host: {
+    'class': 'modal-backdrop fade show',
+    '[@modalBackdrop]': 'animationState',
+    '(@modalBackdrop.done)': 'animationStateChanged.next($event)'
+  },
+  animations: [trigger(
+      'modalBackdrop',
+      [transition('void => enter', useAnimation(fadeIn)), transition('enter => exit', useAnimation(fadeOut))])]
```

Comment:
why not use `:enter` and `:leave`?

Reply:
You can't use `:leave` as you need to wait for the animation to complete before removing the element from the DOM, otherwise you won't see the exit animation. `:enter` could maybe be used in place of `void => enter` but they're subtly different as `:enter` is an alias for `void => *` so it would also match `void => exit` (that's not currently a valid state, but I think it's better to be explicit here for future maintenance)